packages/react-native/android/src/main/java/io/invertase/notifee/NotifeeApiModule.java

```diff
@@ -273,10 +273,17 @@ public void requestPermission(Promise promise) {
         .setRequestPermissionCallback(
             (e, aBundle) -> NotifeeReactUtils.promiseResolver(promise, e, aBundle));
 
-    activity.requestPermissions(
-        new String[] {Manifest.permission.POST_NOTIFICATIONS},
-        Notifee.REQUEST_CODE_NOTIFICATION_PERMISSION,
-        this);
+    try {
+      activity.requestPermissions(
+          new String[] {Manifest.permission.POST_NOTIFICATIONS},
+          Notifee.REQUEST_CODE_NOTIFICATION_PERMISSION,
+          this);
+    } catch (Exception e) {
+      Logger.d(
+          "requestPermission",
+          "Failed to request POST_NOTIFICATIONS permission: " + e.getMessage());
+      NotifeeReactUtils.promiseResolver(promise, e);
```

Comment:
I wanted to make sure so I checked and it looks like this does exactly what we want - e != null so this will reject:

https://github.com/invertase/notifee/blob/6b06c9f3ced0388da99dea061be5e3a1536588f7/packages/react-native/android/src/main/java/io/invertase/notifee/NotifeeReactUtils.java#L100-L107